src/org/sosy_lab/cpachecker/cpa/smg/SMGPredicateManager.java

```diff
@@ -91,13 +95,38 @@ private BooleanFormula addPredicateToFormula(
     BooleanFormula result;
     BigInteger explicitValue = pRelation.getExplicitValue().getValue();
     SMGType symbolicSMGType = pRelation.getSymbolicSMGType();
-    long explicitSize = symbolicSMGType.getCastedSize();
-    boolean isExplicitSigned = symbolicSMGType.isCastedSigned();
+
+    long explicitSize = symbolicSMGType.getCastedSizeLast();
+    boolean isExplicitSigned = symbolicSMGType.getCastedSignedLast();
+
     BinaryOperator op = pRelation.getOperator();
+    BitvectorFormula explicitValueFormula;
+    BitvectorFormula explicitValueFormulaCasted;
+
+    if (explicitValue.compareTo(BigInteger.valueOf(0)) < 0) {
+      if (explicitValue.abs().compareTo(BigInteger.valueOf(1).shiftLeft((int) (explicitSize - 1)))
+          > 0) {
+        symbolicSMGType = new SMGType(new SMGType(64, true), symbolicSMGType);
+        explicitSize = 64;
```

Comment:
I'm a bit confused whether this code block could be correct? For example long double requires 16 bytes memory on x64 architecture.
Please rewrite it and provide additional tests at  test/programs/smg_predicates/